packages/api/lambdas/cleanExecutions.js

```diff
@@ -1,48 +1,248 @@
+      if (cleanupOnlyRunning) queryBuilder.where('status', '=', 'running');
+      else if (cleanupOnlyNonRunning) queryBuilder.where('status', '!=', 'running');
+    })
+    .limit(updateLimit);
+
+  // this is done as a search:update because postgres doesn't support limited updates
+  await knex(executionModel.tableName)
+    .whereIn('cumulus_id', executionIds.map((execution) => execution.cumulus_id))
+    .update(wipedPayloads);
+};
+
+/**
+ * parse out environment variable configuration
+ * @returns {{
+ *   cleanupNonRunning: boolean,
+ *   cleanupRunning: boolean,
+ *   cleanupPostgres: boolean,
+ *   cleanupES: boolean,
+ *   payloadTimeout: number
+ *   esIndex: string,
+ *   updateLimit: number,
+ * }}
+ */
+const parseEnvironment = () => {
+  const cleanupNonRunning = JSON.parse(process.env.CLEANUP_NON_RUNNING || 'true');
+  const cleanupRunning = JSON.parse(process.env.CLEANUP_RUNNING || 'false');
+
+  const cleanupPostgres = JSON.parse(process.env.CLEANUP_POSTGRES || 'true');
+  const cleanupES = JSON.parse(process.env.CLEANUP_ES || 'true');
+  if (!cleanupRunning && !cleanupNonRunning) throw new Error('running and non-running executions configured to be skipped, nothing to do');
+  if (!cleanupES && !cleanupPostgres) throw new Error('elasticsearch and postgres executions configured to be skipped, nothing to do');
+
+  const _payloadTimeout = process.env.PAYLOAD_TIMEOUT || '10';
+  const payloadTimeout = Number.parseInt(_payloadTimeout, 10);
+  if (!Number.isInteger(payloadTimeout)) {
+    throw new TypeError(`Invalid number of days specified in configuration for payloadTimeout: ${_payloadTimeout}`);
+  }
+  const esIndex = process.env.ES_INDEX || 'cumulus';
+
+  const updateLimit = Number(process.env.UPDATE_LIMIT || 10000);
+  return {
+    cleanupRunning,
+    cleanupNonRunning,
+    cleanupPostgres,
+    cleanupES,
+    payloadTimeout,
+    esIndex,
+    updateLimit,
+  };
+};
 
-  const execution = new ExecutionModel();
-  return await execution.removeOldPayloadRecords(completeTimeout,
-    nonCompleteTimeout,
-    completeDisable,
-    nonCompleteDisable);
+/**
+ * parse environment variables to extract configuration and run cleanup of PG and ES executions
+ *
+ * @returns {Promise<void>}
+ */
+async function cleanExecutionPayloads() {
+  const envConfig = parseEnvironment();
+  log.info(`running cleanExecutions with configuration ${JSON.stringify(envConfig)}`);
+  const {
+    updateLimit,
+    cleanupRunning,
+    cleanupNonRunning,
+    cleanupPostgres,
+    cleanupES,
+    payloadTimeout,
+    esIndex,
+  } = envConfig;
+
+  const promises = [];
+  if (cleanupES) {
+    promises.push(cleanupExpiredESExecutionPayloads(
+      payloadTimeout,
+      cleanupRunning,
+      cleanupNonRunning,
+      updateLimit,
+      esIndex
+    ));
+  }
+  if (cleanupPostgres) {
+    promises.push(cleanupExpiredPGExecutionPayloads(
+      payloadTimeout,
+      cleanupRunning,
+      cleanupNonRunning,
+      updateLimit
+    ));
```

Comment:
This approach makes sense to me. If we need to iterate on the PG portion of this in the future, we can do that but still deliver the ES cleanup 👍

Reply:
re my comment above about sleeping during ES cleanup: if the ES client takes longer than the Lambda timeout we never clean up postgres records... is that ok? I know there was a lot of discussion around this so maybe I'm just missing context.

Reply:
Yeah, I think that's a risk. If there's concern that the ES deletion might exceed the Lambda timeout it feels like this should be treated as an async operation.

Reply:
I intended these to be run asynchronously. the two separate promises are not awaited and then we await promise.all on 227, this should (and from my testing does) run these two in parallel

Reply:
and secondly: I think that is OK, if not ideal